app/src/common/shared/com/igalia/wolvic/ui/widgets/NavigationBarWidget.java

```diff
@@ -663,71 +663,73 @@ public void onSessionChanged(@NonNull Session aOldSession, @NonNull Session aSes
         mAttachedWindow.setIsFullScreen(false);
     }
 
-    private void onEnterFullScreen(@NonNull WindowWidget aWindow) {
-        enterFullScreenMode();
-
-        mBeforeFullscreenPlacement = mWidgetPlacement.clone();
-        mWidgetPlacement.cylinder = SettingsStore.getInstance(getContext()).isCurvedModeEnabled();
-        updateWidget();
-
-        if (mAttachedWindow.isResizing()) {
-            exitResizeMode(ResizeAction.KEEP_SIZE);
-        }
-        AtomicBoolean autoEnter = new AtomicBoolean(false);
-        if (getSession().getFullScreenVideo() == null) {
-            mAutoSelectedProjection = VIDEO_PROJECTION_NONE;
-            autoEnter.set(false);
-        } else {
-            mAutoSelectedProjection = VideoProjectionMenuWidget.getAutomaticProjection(getSession().getCurrentUri(), autoEnter);
-        }
+    private void createProjectionMenuIfNeeded() {
+        if (mProjectionMenu != null)
+            return;
 
-        if (mAutoSelectedProjection != VIDEO_PROJECTION_NONE && autoEnter.get()) {
-            mViewModel.setAutoEnteredVRVideo(true);
-            postDelayed(() -> enterVRVideo(mAutoSelectedProjection), 300);
-        } else {
-            mViewModel.setAutoEnteredVRVideo(false);
-            if (mProjectionMenu != null) {
-                mProjectionMenu.setSelectedProjection(mAutoSelectedProjection);
+        mProjectionMenu = new VideoProjectionMenuWidget(getContext());
+        mProjectionMenu.setParentWidget(this);
+        mProjectionMenuPlacement = new WidgetPlacement(getContext());
+        mWidgetManager.addWidget(mProjectionMenu);
+        mProjectionMenu.setDelegate((projection)-> {
+            if (mViewModel.getIsInVRVideo().getValue().get()) {
+                if (projection == VIDEO_PROJECTION_NONE) {
+                    exitVRVideo();
+                    mAttachedWindow.reCenterFrontWindow();
+                } else {
+                    // Reproject while reproducing VRVideo
+                    mWidgetManager.showVRVideo(mAttachedWindow.getHandle(), projection);
+                }
+                closeFloatingMenus();
+            } else {
+                enterVRVideo(projection);
             }
-        }
-        mAttachedWindow.reCenterFrontWindow();
+        });
     }
 
     @Override
-    public void onFullScreen(@NonNull WindowWidget aWindow, boolean aFullScreen) {
+    public void onMediaFullScreen(@NonNull WMediaSession mediaSession, boolean aFullScreen) {
         if (aFullScreen) {
-            if (getSession().getFullScreenVideo() != null) {
-                onEnterFullScreen(aWindow);
+            // The content fullscreen event might have arrived before the media fullscreen event
+            if (!mAttachedWindow.isFullScreen())
+                enterFullScreenMode();
+
+            createProjectionMenuIfNeeded();
```

Comment:
Let's move `createProjectionMenuIfNeeded()` inside the end of `enterFullScreenMode()` (https://github.com/Igalia/wolvic/commit/a0493b1233122755328373e95d6bd122f30bcac3) to fix the bug:

1. Relaunch Wolvic. (make sure mProjectionMenu is null)
2. Go to a website that doesn't trigger auto-play on full screen. (should be anything but YouTube as of Wolvic's plugin) (this won't trigger onMediaFullScreen since by then the video is unavailable)
3. Click on the VR button in the full-screen navigation bar, you will find it unclickable.
4. Start to play video.
5. You will find the VR button in the full-screen navigation bar still unclickable, the only way is to re-enter the full screen.

https://github.com/Igalia/wolvic/assets/43995067/62917c15-3b07-4eb8-9380-1395e85a0634

Reply:
Well that was kind of intentional because I was trying to avoid creating the menu for 2D videos. I even had some code to actually hide the icon. In the end I didn't like it (let's allow users to select whatever projection they want even if it doesn't look good) and decided to revert those changes. The call inside onMediaFullscreen is a leftover of that failed attempt.

Reply:
Actually we won’t be sure about the video format (the YouTube plugin for projection auto selection doesn’t always work, even if we have mozVideoProjection in the URL, we should still be careful that auto play for next video might not remove that parameter), so definitely we should allow users to select no matter what circumstances we have.